cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -390,12 +390,10 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
     });
 
     SelfPacedDueDateEditor = AbstractEditor.extend({
-        fieldName: 'due_num_weeks',
         templateName: 'self-paced-due-date-editor',
         className: 'modal-section-content has-actions due-date-input grading-due-date',
-
         events: {
-            'click .clear-date': 'clearValue',
+            'change #due_in': 'validateDueIn',
             'keyup #due_in': 'validateDueIn',
             'blur #due_in': 'validateDueIn',
```

Comment:
I think I need all 3! The user can change the input 2 ways: 1) with a manual clicking of the arrow in the input (since it is a number input there are up and down arrows when you hover over the input) and 2) keyboard input. (1) is handled with `change` and (2) is handled with `keyup.` `blur` is still good to have I think to validate even when the user clicks away from the input!